storagemanager/storage/views.py

```diff
@@ -23,49 +23,65 @@ class ShowAllListView(ListView):
     model = Item
     template_name = 'storage/show_all.html'
     context_object_name = 'all_items'
-    ordering = ['category', 'type']
+    ordering = ['category', 'type', 'model']
 
 def show_items_from_category(request):
     categories_set = set()                      
     for item in Item.objects.all():
         categories_set.add(item.category)
-    sorted_items = sorted(Item.objects.filter(category=request.POST.get('category')), key=attrgetter('type', 'model'))
+    sorted_items = Item.objects.filter(category=request.POST.get('category')).order_by('type', 'model')
     context = {
         'items_to_show': sorted_items, 
         'categories': categories_set
         }
     return render(request, 'storage/show_items_from_category.html', context)
 
 def delete_item(request):
-    context = {'all_items': sorted(Item.objects.all(), key=attrgetter('category', 'type', 'model'))}
+    context = {'all_items': Item.objects.all().order_by('category', 'type', 'model')}
     return render(request, 'storage/delete_item.html', context)
 
 def delete_item_confirm(request):
-    try:
-        context = {
-            'item_to_delete': str(Item.objects.filter(id=request.POST.get('id')).first()), 
-            'id': request.POST.get('id')
-            }
-        if context.get('item_to_delete').startswith('None'):
-            messages.warning(request, f'Item with that ID could not be found.')
-            return redirect('delete-item')
-        else:
-            return render(request, 'storage/delete_item_confirm.html', context)
-    except ValueError:
+    if request.POST.get('id') == '':
         messages.warning(request, f'Please type a number')
         return redirect('delete-item')
+    elif 'e' in request.POST.get('id'):
+        messages.warning(request, f'Exponent numbers are not allowed')
+        return redirect('delete-item')
+    else:
+        item_id = request.POST.get('id')
+
+    if item_id == 'Choose_item':
+        messages.warning(request, f'Please choose an item from the list.')
+        return redirect('delete-item')
+
+    try:
+        item_to_delete = Item.objects.get(id=item_id)
+    except ObjectDoesNotExist:
+        item_to_delete = None
+    context = {
+        'item_to_delete': item_to_delete, 
+        'id': item_id
+        }
+    if context.get('item_to_delete') == None:
```

Comment:
You are using `get` but you already have access to `item_to_delete` variable, you can just use it :)

Reply:
Still valid comment :)